Importccl: add feature and error telemetry to IMPORT

dt opened the pull request:
This captures telemetry on what file formats and bucketed file size, row count and file count are used in IMPORT, as well as marking many of the errors as Unimplemented errors with unique call-site identifiers to make them easier to track.

dt committed:
telemetry: add bucketing helper

This adds a reusable helper for bucketing numeric telemetry to power of 10.
This should still give us a ballpark idea for what usage looks like — 1 vs 1k etc is useful to know — without being quite as revealing as the raw numbers.

Release note: none.

dt committed:
importccl: mark errors returned by unsupported IMPORT features

This uses the appropriate error code and records the unimplemented feature in telemetry.

Release note: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/28726)
<!-- Reviewable:end -->

dt committed:
importccl: add telemetry for format and file count

Release note (general change): include basic stats about size of IMPORTs run in anonymous telemetry.

rolandcrosby commented:
I'll let @mjibson comment on the correctness of the code, but the data points you're capturing appear to match my expectations. Thanks @dt!

mjibson commented on pkg/ccl/importccl/import_stmt.go:
```diff
@@ -167,19 +169,19 @@ func MakeSimpleTableDescriptor(
 			def.Table.TableNameReference = &n
 		default:
-			return nil, errors.Errorf("unsupported table definition: %s", tree.AsString(def))
+			return nil, pgerror.Unimplemented(fmt.Sprintf("import.%T", def), "unsupported table definition: %s", tree.AsString(def))
```
Can this leak user info like column names?

dt replied:
no, the only reported thing is the first field, the feature code, which in this case is `import.%T` so it can only be the type of an AST node

mjibson commented on pkg/sql/pgwire/pgerror/errors.go:
```diff
@@ -151,14 +151,14 @@ func UnimplementedWithIssueError(issue int, msg string) error {
-func UnimplementedWithDepth(depth int, feature, msg string) *Error {
-	err := NewErrorWithDepthf(depth+1, CodeFeatureNotSupportedError, "%s", msg)
+func UnimplementedWithDepth(depth int, feature, msg string, args ...interface{}) *Error {
+	err := NewErrorWithDepthf(depth+1, CodeFeatureNotSupportedError, msg, args...)
```
This has the possibility to break stuff if `%` was used in any of the Unimplemented messages. Do we need to audit current use to make sure it's ok?

dt replied:
it isn't -- there are a couple Sprintf'ers that could migrate to this, but I don't want to go refactoring right now.

dt commented:
bors r+
